CHANGELOG.md

```diff
@@ -0,0 +1,215 @@
+   a different LSM mount context on restore. (#3068)
+ * runc releases now are cross-compiled for several architectures and static
+   builds for said architectures are now available for all future releases.
+   (#3197)
+ * intelrdt: support ClosID parameter. (#2920)
+ * runc exec --cgroup: an option to specify a (non-top) in-container cgroup
+   to use for the process being executed. (#3040 #3059)
+ * cgroup v1 controllers now support hybrid hierarchy (i.e. when on a cgroup v1
+   machine a cgroup2 filesystem is mounted to /sys/fs/cgroup/unified, runc
+   run/exec now adds the container to the appropriate cgroup under it). (#2087
+   #3059)
+ * sysctl: allow slashes in sysctl names, to better match `sysctl(8)`'s
+   behaviour. (#3254 #3257)
+ * mounts: when using user namespaces, open bind-mount sources from the host
+   namespace and use those file descriptors as the mount sources to allow runc
+   to create bind-mounts to paths that the container userns cannot access after
+   switching. (#2576)
+ * Add support for recursive mount attributes using `mount_setattr(2)`. These
+   have the same names as the proposed `mount(8)` options -- just prepend `r`
+   to the option name (such as `rro`). (#3272)
+ * Add `runc features` subcommand to allow runc users to detect what features
+   runc has been built with. This includes critical information such as
+   supported mount flags, hook names, and so on. Note that the output of this
+   command is subject to change and will not be considered stable until runc
+   1.2 at the earliest. (#3296)
+
+### Changed
+ * system: improve performance of `/proc/$pid/stat` parsing. (#2696)
+ * cgroup2: when `/sys/fs/cgroup` is configured as a read-write mount, change
+   the ownership of certain cgroup control files (as per
+   `/sys/kernel/cgroup/delegate`) to allow for proper defferal to the container
+   process. (#3057)
+
+#### libcontainer API
+ * internal api: remove internal error types and handling system, switch to Go
+   wrapped errors. (#3033)
+ * New configs.Cgroup structure fields (#3177):
+   * Systemd (whether to use systemd cgroup manager); and
+   * Rootless (whether to use rootless cgroups).
+ * New cgroups/manager package aiming to simplify cgroup manager instantiation.
+   (#3177)
+ * All cgroup managers' instantiation methods now initialize cgroup paths and
+   can return errors. This allows to use any cgroup manager method (e.g.
+   Exists, Destroy, Set, GetStats) right after instantiation, which was not
+   possible before (as paths were initialized in Apply only). (#3178)
+
+### Fixed
+ * nsenter: do not try to close already-closed fds during container setup and
+   bail on close(2) failures. (#3058)
+ * runc checkpoint/restore: fixed for containers with an external bind mount
+   which destination is a symlink. (#3047).
+ * cgroup: improve openat2 handling for cgroup directory handle hardening.
+   (#3030)
+ * `runc delete -f` now succeeds (rather than timing out) on a paused
```

Comment:
`s/timing out/timeouts/`

Or "runc is now succeeding (rather than timing out)"...

Reply:
Timeouts doesn't sound right to me -- "timeouts" is a plural, it's not the same thing as "times out" -- that's why I changed it from the original changelog.

Reply:
If you don't like "timeouts" as a verb, use "times out". In the current form, we're mixing present simple ("succeeds") and present continuous ("timing out"), and it also feels like "is" is missing.

So, I suggest either

* `runc delete -f` now succeeds (rather than times out) on a paused ...

or

* `runc delete -f` is now succeeding (rather than timing out) on a paused ...

Reply:
To be honest (as a native speaker) all of the options you've listed sound wrong as complete sentences. I think it's because of "rather than" being included in the mix -- if I'd written "succeeds or timing out" you would be entirely correct saying it's wrong (and the two choices you listed would be correct -- "succeeds or times out" or "succeeding or timing out" -- though in this context the first fits better) but that's not the sentence.

Reply:
Right, as of this morning I agree that "rather than" changes the structure; thanks for the explanation!